lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -0,0 +1,131 @@
+package com.kmwllc.lucille.tika.stage;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.util.FileUtils;
+import com.typesafe.config.Config;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.BodyContentHandler;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+/**
+ * This stage uses Apache Tika to perform text and metadata extraction
+ * <br>
+ * Config Parameters -
+ * <br>
+ * text_field (String, Optional) : name of destination field for parsed data to be placed
+ * file_path_field (String, Optional) : name of field from which file path can be extracted, if file_path_field and byte_array_field both not provided, stage will do nothing
+ * byte_array_field (String, Optional) : name of field from which byte array data can be extracted
+ * tika_config_path (String, Optional) : path to tika config, if not provided will default to empty AutoDetectParser
+ * metadata_prefix (String, Optional) : prefix to be appended to fields for metadata information extracted after parsing
+ */
+public class TextExtractor extends Stage {
+
+  private static final Logger log = LogManager.getLogger(TextExtractor.class);
+  private String textField;
+  private String filePathField;
+  private String tikaConfigPath;
+  private String byteArrayField;
+  private String metadataPrefix;
+  private Parser parser;
+  private ParseContext parseCtx;
+
+  public TextExtractor(Config config) {
+    super(config, new StageSpec().withOptionalProperties("text_field", "file_path_field", "byte_array_field", "tika_config_path",
+        "metadata_prefix"));
+    textField = config.hasPath("text_field") ? config.getString("text_field") : "text";
+    filePathField = config.hasPath("file_path_field") ? config.getString("file_path_field") : "filepath";
+    byteArrayField = config.hasPath("byte_array_field") ? config.getString("byte_array_field") : "byte_array";
+    metadataPrefix = config.hasPath("metadata_prefix") ? config.getString("metadata_prefix") : "tika";
```

Comment:
Thanks for sharing the example with the metadata @abijitran. I think we will very quickly find that we want to be able to 
  1. suppress certain metadata from being extracted with a whitelist or blacklist and also 
  2. to have an ability to rename certain metadata fields to allow users to make sure that tika doesn't produce fields that will result in schema mismatch errors. (I'm not sure if the mapping is something we would defer to other stage(s) or if we want the tika stage to be self-sufficient. @rseitz do you have an opinion on this?)

Reply:
1. Sure, will add this functionality soon
2. There exists a tika_prefix config property that should prevent schema mismatch errors (am going to support that to be empty so that users can have no prefix as well). Talking with Rudi, since RenameFields already handles renaming, we're thinking that it's best to leave that role to that stage.